src/frontend/apps/impress/src/features/docs/doc-export/utils.ts

```diff
@@ -293,6 +293,170 @@ ${editorHtmlWithLocalMedia}
 </html>`;
 };
 
+/**
+ * Enrich the HTML produced by the editor with semantic tags and basic a11y defaults.
+ *
+ * Notes:
+ * - We work directly on the parsed Document so modifications are reflected before we zip files.
+ * - We keep the editor inner structure but upgrade the key block types to native elements.
+ */
+export const improveHtmlAccessibility = (
+  parsedDocument: Document,
+  documentTitle: string,
+) => {
+  const body = parsedDocument.body;
+  if (!body) {
+    return;
+  }
+
+  // 1) Headings: convert heading blocks to h1-h6 based on data-level
+  const headingBlocks = Array.from(
+    body.querySelectorAll<HTMLElement>("[data-content-type='heading']"),
+  );
+
+  headingBlocks.forEach((block) => {
+    const rawLevel = Number(block.getAttribute('data-level')) || 1;
+    const level = Math.min(Math.max(rawLevel, 1), 6);
+    const heading = parsedDocument.createElement(`h${level}`);
+    heading.innerHTML = block.innerHTML;
+    block.replaceWith(heading);
+  });
+
+  // 2) Lists: group consecutive list items into UL/OL with LI children
+  const listItemSelector =
+    "[data-content-type='bulletListItem'], [data-content-type='numberedListItem']";
+  const listItems = Array.from(
+    body.querySelectorAll<HTMLElement>(listItemSelector),
+  );
+
+  listItems.forEach((item) => {
+    const parent = item.parentElement;
+    if (!parent) {
+      return;
+    }
+
+    const isBullet =
+      item.getAttribute('data-content-type') === 'bulletListItem';
+    const listTag = isBullet ? 'ul' : 'ol';
+
+    // If the previous sibling is already the right list, reuse it; otherwise create a new one.
+    let previousSibling = item.previousElementSibling;
+    let listContainer: HTMLElement | null = null;
+
+    if (previousSibling?.tagName.toLowerCase() === listTag) {
+      listContainer = previousSibling as HTMLElement;
+    } else {
+      listContainer = parsedDocument.createElement(listTag);
+      parent.insertBefore(listContainer, item);
+    }
+
+    const li = parsedDocument.createElement('li');
+    li.innerHTML = item.innerHTML;
+    listContainer.appendChild(li);
+    parent.removeChild(item);
+  });
+
+  // 3) Quotes -> <blockquote>
+  const quoteBlocks = Array.from(
+    body.querySelectorAll<HTMLElement>("[data-content-type='quote']"),
+  );
+  quoteBlocks.forEach((block) => {
+    const quote = parsedDocument.createElement('blockquote');
+    quote.innerHTML = block.innerHTML;
+    block.replaceWith(quote);
+  });
```

Comment:
We have a recession on that:
Before: 
<img width="959" height="271" alt="Image" src="https://github.com/user-attachments/assets/83edc315-47ae-4e58-95ef-f4982dcd24a6" />

After:

<img width="526" height="184" alt="Image" src="https://github.com/user-attachments/assets/3f1189d3-7ef0-4295-a62e-4604d3805c14" />